include/revng/Pipeline/Step.h

```diff
@@ -0,0 +1,387 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <memory>
+#include <set>
+#include <string>
+#include <type_traits>
+#include <utility>
+#include <vector>
+
+#include "llvm/ADT/ArrayRef.h"
+#include "llvm/ADT/SmallVector.h"
+#include "llvm/ADT/StringRef.h"
+#include "llvm/ADT/iterator_range.h"
+#include "llvm/Support/Error.h"
+
+#include "revng/ADT/STLExtras.h"
+#include "revng/Pipeline/Containers.h"
+#include "revng/Pipeline/Context.h"
+#include "revng/Pipeline/InputOutputContract.h"
+#include "revng/Support/Debug.h"
+#include "revng/Support/FunctionIntrospection.h"
+
+namespace Pipeline {
+
+template<typename T>
+class NamedPtr {
```

Comment:
* [ ] No abbreviation
* [ ] What's the purpose of this class?

Reply:
well, not even for the ptr like objects? unique_ptr, shared_ptr?

Reply:
I'd suggest we violate our own rules only for the sake of coherency with stdlib.
For instance, type names have first letter capital, but if you need to use `using value_type = ...`, you use `value_type`.
Here the questions is: are we trying to mimic/be compatible with some part of the stdlib? If so, OK, but then go for `named_ptr`, otherwise, I'd go for explicit.